packages/evm/contracts/Types.sol

```diff
@@ -47,16 +47,16 @@ enum Operator {
     /* 13: */ _ComplexPlaceholder13,
     /* 14: */ _ComplexPlaceholder14,
     /* 15: */ _ComplexPlaceholder15,
-    /* 16: */ _ComplexPlaceholder16,
     // ------------------------------------------------------------
-    // 17-31: COMPARISON EXPRESSIONS
+    // 16-31: COMPARISON EXPRESSIONS
     //          paramType: Static / Dynamic / Tuple / Array / AbiEncoded
-    //          🚫 children
+    //          ❓ children (optional, used for describing the structure of complex types)
     //          ✅ compValue
-    /* 17: */ EqualTo,
-    /* 18: */ GreaterThan,
-    /* 19: */ LessThan,
-    /* 20: */ Bitmask,
+    /* 16: */ EqualTo,
+    /* 17: */ GreaterThan, // only for paramType: Static (uint)
+    /* 18: */ LessThan,    // only for paramType: Static (uint) 
+    /* 19: */ Bitmask,
+    /* 20: */ _ComparisonPlaceholder20,
     /* 21: */ _ComparisonPlaceholder21,
```

Comment:
we might want to add `<` and `>` for signed int in the future. shall we already add some placeholders for these?

```suggestion
    /* 17: */ Bitmask,
    /* 18: */ GreaterThan, // only for paramType: Static (uint)
    /* 19: */ LessThan,     // only for paramType: Static (uint) 
    /* 20: */ _SignedGreaterThanPlaceholder,  // only for paramType: Static (int)
    /* 21: */ _SignedLessThanPlaceholder,        // only for paramType: Static (int)
```

Reply:
Or maybe even include `uint` in the name, like we do for arrays?

- `UintGreaterThan`
- `IntGreaterThan`

Reply:
hummmm. this is not too important I would think

Reply:
obviously not important at all. but let's just make it 100% water- and future-proof.

Reply:
In general, it's hard to future proof stuff

Reply:
then how about only renaming `GreaterThan` and `LessThan` to `UintGreaterThan` and `UintLessThan`? It's matching the `Array...` operators names and brings some extra clarity

Reply:
I would do the reordering before I do this renaming.

Reason: we make nowhere in our code any assumption about what is in inside a Static data type. We don't do in the Decoder, and thus we also won't do it at the operator level. For starters, we could also do signed comparisons, but I'd say this operator is free to be used at will with any Static type.

Reply:
ok

I just feel, it's currently not obvious enough to users that using `GreaterThan` and `LessThan` will have unexpected effect when used on `int`. We wanna avoid people using it wrong so accidentally opening back door in their permissions.